gemini/function-calling/function_calling_data_structures.ipynb

```diff
@@ -0,0 +1,745 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #4.    def recurse_proto_repeated_composite(repeated_object):
</pre><p>There aren't any built in functions for this? I'm surprised, this should probably be added to the proto-plus library or the Vertex AI Sdk</p><p> </p><p>Can you add typing for the parameters (where feasible) and the return types?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/507/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
No, not in the Gemini SDKs directly. We're in the middle of some longer discussions about making this better in the SDKs. And, normally I wouldn't need to use this and I could grab the function name and params. But in this notebook we go deeper into more complex schemas where I needed these helpers. 😅